common/remote/sockfile.cpp

```diff
@@ -1114,7 +1163,19 @@ protected: friend class CRemoteFileIO;
                 }
             }
             if (!socket) {
-                connectSocket(tep);
+                if (useSSL) {
+                    try {
+                        connectSocket(tep);//first try secure connect
+                    }
+                    catch(...) {
+                        useSSL = false;
+                        tep.port = DAFILESRV_PORT;
+                        PROGLOG("Secure connect failed, retrying on legacy port");
+                        connectSocket(tep);
```

Comment:
I'm not sure it's safe if this throw an exception inside the catch.
I think you can simplify by putting 1st connect(tep) in a catch, then throwing if !useSSL or continuing to try 2nd beyond try/catch if set.